hid-uclogic-params.c

```diff
@@ -830,12 +854,17 @@ static int uclogic_params_huion_init(struct uclogic_params *params,
 	bInterfaceNumber = iface->cur_altsetting->desc.bInterfaceNumber;
 
 	/* If it's a custom keyboard interface */
-	if (bInterfaceNumber == 1) {
+	/*
+	 * The pen interface now is not limit to 0, such as KD200, it has
+	 * two pen interfaces. It seems this thing also occurs in the hardware
+	 * with device id 0x006d.
+	 */
+	if (bInterfaceNumber == 1 || bInterfaceNumber == 2) {
 		/* Keep everything intact, but mark pen usage invalid */
 		p.pen.usage_invalid = true;
 		goto output;
 	/* Else, if it's not a pen interface */
-	} else if (bInterfaceNumber != 0) {
+	} else if (bInterfaceNumber > 2) {
```

Comment:
> @inochisa thanks a lot for the quick reply.
> 
> I'm trying to understand this second patch a bit better.
> 
> What are the events sent by the interface with `bInterfaceNumber == 2`? The keyboard events? In that case, it make sense to add `if (bInterfaceNumber == 1 || bInterfaceNumber == 2)` so the interface is used but not as a pen.
> 

Yes, it is keyboard events.

> About the second condition. Can this `else if` condition remain unmodified?
> 

I think this can be left unmodified. As we already check `bInterfaceNumber` "1" and "2", it is OK to use existed check.

> I made a minor modification of your patch in the PR I'm working on for v13 to illustrate what I mean. I think it should work: [0462d43](https://github.com/DIGImend/digimend-kernel-drivers/commit/0462d43122f18a73e4c8497235c9138d5e676b37)

Thanks

Reply:
Merged! Thanks a lot :tada: 

You can rebase this PR on top of master if you want. But I'm trying to avoid refactors without KUnit tests to ensure the functionality is kept.

About the firmware list... I recently discovered that they are unique and that can be used to identify the tablets in user-space https://github.com/linuxwacom/libwacom/pull/659

It might be a good idea to add a full list of firmwares in the future, but for the moment I'd like to keep it as it is to avoid bugs/regressions until I have a clear idea of how we can take advantage of it.

Reply:
> Merged! Thanks a lot 🎉
> 
> You can rebase this PR on top of master if you want. But I'm trying to avoid refactors without KUnit tests to ensure the functionality is kept.
> 

I prefer to close this PR as the left in this patch is hard to evolve.

> About the firmware list... I recently discovered that they are unique and that can be used to identify the tablets in user-space [linuxwacom/libwacom#659](https://github.com/linuxwacom/libwacom/pull/659)
> 
> It might be a good idea to add a full list of firmwares in the future, but for the moment I'd like to keep it as it is to avoid bugs/regressions until I have a clear idea of how we can take advantage of it.

Excellent. I see your tool detect the feather of KD200. I will check result and see there is anything that can improve/